wgpu-core/src/instance.rs

```diff
@@ -242,6 +251,48 @@ impl From<HalDeviceType> for DeviceType {
     }
 }
 
+impl PartialEq<HalDeviceType> for DeviceType {
```

Comment:
Why do we need this, as opposed to converting the `HalDeviceType` to `DeviceType` and using an existing `PartialEq` implementation?

Reply:
I just didn't like to clone here, `gfx_hal::adapter::DeviceType` just doesn't implement `Copy`, in any case, I just used clone now.